packages/components/src/styles/components/modal.scss

```diff
@@ -8,16 +8,18 @@
 //
 
 .hds-modal {
+  position: fixed;
   z-index: 50;
-  flex-direction: column;
+  height: fit-content;
+  margin: auto;
   padding: 0;
   background: var(--token-color-surface-primary);
   border: none;
   border-radius: 8px;
   box-shadow: var(--token-surface-overlay-box-shadow);
+  inset: 0;
```

Comment:
nitpick: I would move it under the `z-index` declarations so it's near the "position/layout" block (they're related)